.github/workflows/publish-helm-chart.yml

```diff
@@ -0,0 +1,90 @@
+          # 4. For everything else (e.g., dispatches), build all charts
+          if [[ $GITHUB_EVENT_NAME == 'pull_request' ]] ||
+             [[ $GITHUB_EVENT_NAME == 'push' ]]; then
+               if [[ $GITHUB_EVENT_NAME == 'pull_request' ]]; then
+                   BASE=$(git merge-base origin/$GITHUB_BASE_REF HEAD)
+               elif [[ $GITHUB_REF == 'refs/heads/main' ]]; then
+                   BASE=${{github.event.before}}
+               else
+                   BASE=origin/main
+               fi
+               # List helm-chart root dirs where files have changed and the
+               # root dir exists. Example value:
+               # "helm-charts/supported helm-charts/contrib"
+               helmcharts=$(git diff --name-only \
+                                 "$BASE" \
+                                 "$GITHUB_SHA" |
+                        egrep "^$ORG_DIR/" |
+                        cut -d/ -f -2 |
+                        sort |
+                        uniq |
+                        xargs -I {} find . -type d \
+                                           -wholename ./{} \
+                                           -printf "%P\n")
+          else
+               # List all image root dirs. Example value:
+               # "helm-charts/supported helm-charts/contrib"
+               helmcharts=$(find $ORG_DIR -mindepth 1 \
+                                      -maxdepth 1 \
+                                      -type d \
+                                      -printf "$ORG_DIR/%P\n")
+          fi
+
+          # Ensure that the generated JSON array has a member,
+          # otherwise GHA will throw an error about an empty matrix
+          # vector in subsequent steps
+          helm_json=$(echo -n "${helmcharts:-dummy}" | jq -Rcs '.|split("\n")')
+          echo "::set-output name=json::$helm_json"
+
+  build-and-push:
+      runs-on: ubuntu-latest
+      needs: build-helm-list
+      if: >-
+        github.ref == 'refs/heads/main' &&
+        github.event_name != 'pull_request' &&
+        github.repository_owner == 'osg-htc' &&
+        needs.build-helm-list.outputs.helmcharts != '["dummy"]'
+      continue-on-error: true
+      steps:
+          - name: Checkout code
+            uses: actions/checkout@v3
+          
+          - name: Install Helm
+            run: |
+              curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3
+              chmod 700 get_helm.sh
+              ./get_helm.sh
+
+          - name: Build Helm Chart
+            run: |
+              echo ${{secrets.OSG_HABOR_PASSWORD}} | helm registry login -u ${{secrets.OSG_HABOR_USER}} --password-stdin hub.opensciencegrid.org/osg-htc
+              helm_package_path=$(helm package supported/opensciencegrid/osdf-origin)
+              helm_package_name=$(basename "$helm_package_path")
+              helm push $helm_package_name oci://hub.opensciencegrid.org/osdf-origin
```

Comment:
A few things:

1. There should be a step before this that attempts a build and this step should either only run on PRs if it duplicates the build that the push command is presumably doing, or run on both pushes/PRs if the push is smart enough to not rebuild if there's an existing build artifact
2. The chart to build is still hardcoded, meaning that this doesn't work when we add additional charts to `supported/osg-htc/`. For example, if we create `supported/osg-htc/jeffs-new-chart/` and merge that, then this will still /only/ build the `osdf-origin` chart
3. This job doesn't actually use the list of changed charts from the previous job meaning that even after you address (2), every merge into `main` will result in a rebuild and push of every chart regardless of it being changed or not. The images repo uses the list from the previous job to dynamically construct the matrix of images that need to be rebuilt and pushed: https://github.com/opensciencegrid/images/blob/main/.github/workflows/build-containers.yml#L107-L110

Reply:
@brianhlin For #1 do we want a step that's like a test build that's only runs if it's the same build as what is being pushed? Just a little confused on exactly what we want happening there.

Reply:
Basically we only want to push to hub.opensciencegrid.org when commits are pushed to the `main` branch on this fork, i.e. the repo is `osg-htc/helm-charts`. However, we still want useful CI (a `helm lint` and `helm package` are attempted) when 1) someone submits a PR against this repo and 2) when someone commits stuff on their own fork. 

Putting those together, we need to split up the package and push into two separate steps with appropriate `if:` expressions.